### PR TITLE
Improve `get_item` and `get_item_by_title` functions

### PR DESCRIPTION
**Pull request by edif2008**
`get_item_by_title` now retrieves the item details instead of the summary.
`get_item` can now accept item id or title, and vault id or name

### CODE REVIEW

File: src/onepasswordconnectsdk/client.py [hunk 1]
```diff
@@ -44,25 +44,53 @@ def create_session(self):
...
         """
 
         headers = {}
         headers["Authorization"] = f"Bearer {self.token}"
         headers["Content-Type"] = "application/json"
         return headers
 
-    def get_item(self, item_id: str, vault_id: str):
+    def get_item(self, item: str, vault: str):
+        """Get a specifiv item
```
Comment (jillianwilson) [inline src/onepasswordconnectsdk/client.py:58]: typo: change `specifiv` to `specify`

File: src/onepasswordconnectsdk/client.py [hunk 2]
```diff
@@ -44,25 +44,53 @@ def create_session(self):
...
+            FailedToRetrieveItemException: Thrown when a HTTP error is returned
+            from the 1Password Connect API
+
+        Returns:
+            Item object: The found item 
+        """
+
+        vault_id = vault
+        if self._is_valid_UUID(vault) is False:
+            vault_id = self.get_vault_by_title(vault).id
```
Comment (jillianwilson) [inline src/onepasswordconnectsdk/client.py:74]: we should handle if no vault is found by title

Comment (edif2008) [reply src/onepasswordconnectsdk/client.py:74]: This is already handled by `get_vault_by_item` function. It throws an error, similar to the one thrown when no items with the given title are found. Check [this snippet](https://github.com/1Password/connect-sdk-python/blob/2799f30aeaf2e9226bc0d935417b471d0bda2aed/src/onepasswordconnectsdk/client.py#L107) for `get_vault_by_title`, compared to [this snippet](https://github.com/1Password/connect-sdk-python/blob/2799f30aeaf2e9226bc0d935417b471d0bda2aed/src/onepasswordconnectsdk/client.py#L272) for `get_item_by_title`

Comment (jillianwilson) [reply src/onepasswordconnectsdk/client.py:121]: Thanks!

**Review by jillianwilson — APPROVED** [review]

**Pull request merged**